pandas/io/parquet.py

```diff
@@ -184,6 +186,11 @@ def write(
 
         table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
 
+        df_metadata = {"df.attrs": json.dumps(df.attrs)}
```

Comment:
maybe `PANDAS_ATTRS` is a more specific key than `"df.attrs"`

Reply:
Also, probably best to guard this with an `if df.attrs:`

Reply:
Changed key name to `PANDAS_ATTRS` and added `if df.attrs` gaurd